include/aikido/control/PositionCommandExecutor.hpp

```diff
@@ -1,29 +1,48 @@
 #ifndef AIKIDO_CONTROL_POSITIONCOMANDEXECUTOR_HPP_
 #define AIKIDO_CONTROL_POSITIONCOMANDEXECUTOR_HPP_
 
+#include <chrono>
 #include <future>
 #include <Eigen/Dense>
 
 namespace aikido {
 namespace control {
 
-/// Abstract class for executing position commands. A position command requests
-/// several degrees of freedom to move to a specified configuration and provides
-/// feedback when the move is complete.
+/// Abstract class for executing position commands.
+///
+/// A position command requests several degrees of freedom to move to a
+/// specified configuration and provides feedback when the move is complete.
+///
+/// \note A move may complete before the DOFs reach the specified configuration
+/// due to e.g. collision.
 class PositionCommandExecutor
 {
 public:
+  /// Constructor.
+  ///
+  /// \param timestep The time period that each call to step() should simulate
+  PositionCommandExecutor(std::chrono::milliseconds timestep);
+
   virtual ~PositionCommandExecutor() = default;
 
-  /// Execute hand to goalPosition
+  /// Move hand to goalPosition.
+  ///
   /// \param goalPositions Goal positions
-  /// \return future which becomes available when movement stops (which may not
-  ///         be when the hand reaches the specified goalPositions due to e.g.
-  ///         collision)
+  /// \return future which becomes available when movement stops
   virtual std::future<void> execute(const Eigen::VectorXd& goalPositions) = 0;
 
   // Step once.
   virtual void step() = 0;
+
+  /// Get the current timestep.
+  virtual std::chrono::milliseconds getTimestep() const;
+
+  /// Set the current timestep.
+  virtual void setTimestep(std::chrono::milliseconds timestep);
```

Comment:
Does this mean that we can arbitrarily change the timestep during execution? Isn't it too confusing? Why not just set it as a const member variable defined at the constructor?